examples/ShoppingAds/AddPerformanceMaxProductListingGroupTree.php

```diff
@@ -0,0 +1,602 @@
+        int $customerId,
+        int $assetGroupId,
+        int $assetGroupListingGroupFilterId,
+        ?int $parentId = null,
+        ?ListingGroupFilterDimension $listingGroupFilterDimension = null
+    ): MutateOperation {
+        $assetGroupListingGroupFilter = new AssetGroupListingGroupFilter([
+            'resource_name' => ResourceNames::forAssetGroupListingGroupFilter(
+                $customerId,
+                $assetGroupId,
+                $assetGroupListingGroupFilterId
+            ),
+            'asset_group' => ResourceNames::forAssetGroup($customerId, $assetGroupId),
+            // Sets the type as a SUBDIVISION, which will allow the node to be the parent of
+            // another sub-tree.
+            'type' => ListingGroupFilterType::SUBDIVISION,
+            // Because this is a Performance Max campaign for retail, we need to specify
+            // that this is in the shopping vertical.
+            'vertical' => ListingGroupFilterVertical::SHOPPING
+        ]);
+        // If $parentId and $listingGroupFilterDimension are not null, creates
+        // a non-root division by setting its parent and case value.
+        if (!is_null($parentId) && !is_null($listingGroupFilterDimension)) {
+            $assetGroupListingGroupFilter->setParentListingGroupFilter(
+                ResourceNames::forAssetGroupListingGroupFilter(
+                    $customerId,
+                    $assetGroupId,
+                    $parentId
+                )
+            );
+            // Case values contain the listing dimension used for the node.
+            $assetGroupListingGroupFilter->setCaseValue($listingGroupFilterDimension);
+        }
+
+        return new MutateOperation([
+            'asset_group_listing_group_filter_operation'
+                => new AssetGroupListingGroupFilterOperation([
+                    'create' => $assetGroupListingGroupFilter
+                ])
+        ]);
+    }
+    // [END add_performance_max_product_listing_group_tree_5]
+
+    /**
+     * Creates a mutate operation that creates a child asset group listing group filter (unit
+     * node).
+     *
+     * Use this method if the filter won't have child filters. Otherwise, use
+     * createMutateOperationForSubdivision().
+     *
+     * @param int $customerId the customer ID
+     * @param int $assetGroupId the asset group ID
+     * @param int $assetGroupListingGroupFilterId the ID of the asset group listing group filter to
+     *     be created
+     * @param int $parentId the ID of the parent of asset group listing group filter to be
+     *      created
+     * @param ListingGroupFilterDimension $listingGroupFilterDimension the listing group
+     *     filter dimension to associate with the asset group listing group filter
+     * @return MutateOperation the mutate operation for creating a unit
+     */
+    // [START add_performance_max_product_listing_group_tree_6]
```

Comment:
ditto: The golden includes the function comment (not sure if intended but let's keep it consistent)